src/components/DateField/DateFieldInt.part.tsx

```diff
@@ -0,0 +1,367 @@
+
+  private changeValue: ReactDatePickerProps['onChange'] = inputDate => {
+    const { dateFormat = defaultDateFormat, locale } = this.props;
+    const date = inputDate || new Date();
+
+    const value = safeDateFormat(date, {
+      dateFormat,
+      locale,
+    });
+
+    this.change(date, value);
+  };
+
+  private changeInput = (e: React.FocusEvent<HTMLInputElement>) => {
+    const { onChangeRaw } = this.props;
+    const { value } = e.target;
+    this.change(this.parseDate(value), value);
+
+    if (typeof onChangeRaw === 'function') {
+      onChangeRaw(e);
+    }
+  };
+
+  private parseDate = (value: string) => {
+    const { locale, dateFormat = defaultDateFormat, strictParsing } = this.props;
+    return parseDate(value, dateFormat, locale, strictParsing) || new Date();
+  };
+
+  private change = (date: Date, value: string) => {
+    const { onChange, name = '', form } = this.props;
+
+    if (!this.valueControlled) {
+      if (form) {
+        form.change({
+          name,
+          value,
+        });
+      } else {
+        this.setState({
+          value,
+          date,
+        });
+
+        this.onOpenChange(true);
+      }
+    }
+
+    if (typeof onChange === 'function') {
+      onChange({
+        value,
+        date,
+      });
+    }
+  };
+
+  private onClickOutside: DateFieldProps['onClickOutside'] = event => {
+    const { onClickOutside } = this.props;
+    this.onOpenChange(false);
+    if (typeof onClickOutside === 'function') {
+      onClickOutside(event);
+    }
+  };
+
+  private onFocus: DateFieldProps['onFocus'] = () => {
+    const { onFocus } = this.props;
+    this.onOpenChange(true);
+    if (typeof onFocus === 'function') {
+      onFocus();
+    }
+  };
+
+  private onBlur: DateFieldProps['onBlur'] = () => {
+    const { onBlur } = this.props;
+    this.onOpenChange(false);
+    if (typeof onBlur === 'function') {
+      onBlur();
+    }
+  };
+
+  render() {
+    const {
+      autoComplete,
+      months,
+      weekDays,
+      open,
+      placeholder,
+      name: _0,
+      value: _1,
+      defaultValue: _2,
+      onChange: _3,
+      onOpenChange: _4,
+      onChange: _5,
+      onClickOutside: _6,
+      onFocus: _7,
+      onBlur: _8,
+      ...rest
+    } = this.props;
+
+    const datePickerProps = {} as ReactDatePickerProps;
+    const otherProps = {} as Omit<DateFieldBasicProps, 'onChange' | 'children'>;
+    Object.keys(rest).forEach(propName => {
+      if (excludedReactDatePickerProps[propName]) {
+        return;
+      }
+
+      if (reactDatepickerProps[propName]) {
+        datePickerProps[propName] = rest[propName];
+      } else {
+        otherProps[propName] = rest[propName];
+      }
+    });
+
+    const { value, error, date } = this.state;
+
+    return (
+      <CustomReactDatepicker
+        customInput={<DatePickerTextField {...otherProps} error={error} autoComplete={autoComplete} />}
+        placeholderText={placeholder}
+        customInputRef="inputRef"
```

Comment:
what is the purpose of this prop?

Reply:
As one of the props `react-datepicker` takes React.Element as `customInput `. In our case `customInput ` is our `TextField` component.  `react-datepicker` needs to get access to the original input field. It's possible to pass ref to the `TextField` through the prop `inputRef`. Through `customInputRef ` I tell to `react-datepicker` to use `inputRef` instead of `ref` prop (which is the default and doesn't work for `TextField` )